upgrade_downgrade/common.sh

```diff
@@ -110,72 +115,62 @@ _waitForPodsReady() {
   return 1
 }
 
-waitForPodsReady() {
+function wait_for_pods_ready() {
   echo "Waiting for pods to be ready in ${1}..."
-  withRetriesMaxTime 600 10 _waitForPodsReady "${1}"
+  with_retries_max_time 900 10 _wait_for_pods_ready "${1}"
   echo "All pods ready."
 }
 
-waitForJob() {
-  echo "Waiting for job ${1} to complete..."
-  local start_time=${SECONDS}
-  until kubectl get jobs -n "${2}" "${1}" -o jsonpath='{.status.conditions[?(@.type=="Complete")].status}' | grep True ; do
-      sleep 1 ;
-  done
-  run_time=0
-  (( run_time = SECONDS - start_time ))
-  echo "Job ${1} ran for ${run_time} seconds."
-}
-
-restartDataPlane() {
+function restart_data_plane() {
   local name="$1"
   local namespace="$2"
-  writeMsg "Restarting deployment ${namespace}/${name}"
+  write_msg "Restarting deployment ${namespace}/${name}"
   echo_and_run_or_die kubectl rollout restart deployment "${name}" -n "${namespace}"
   echo_and_run_or_die kubectl rollout status deployment "${name}" -n "${namespace}" --timeout=30m  
 }
 
-# Return 1 if the specific error code percentage exceed corresponding threshold
-errorPercentBelow() {
-  local LOG=${1}
-  local ERR_CODE=${2}
-  local LIMIT=${3}
-  local s
-  s=$(grep "Code ${ERR_CODE}" "${LOG}")
-  local regex="Code ${ERR_CODE} : [0-9]+ \\(([0-9]+)\\.[0-9]+ %\\)"
-  if [[ ${s} =~ ${regex} ]]; then
-    local pctErr="${BASH_REMATCH[1]}"
-    if (( pctErr > LIMIT )); then
-      return 1
-    fi
-    echo "Errors percentage is within threshold"
-  fi
-  return 0
-}
-
 # Make a copy of test manifests in case either to/from branch doesn't contain them.
-copy_test_files() {
+function copy_test_files() {
   rm -Rf "${TMP_DIR}"
   mkdir -p "${TMP_DIR}"
   echo "${WD}"
-  cp -f "${WD}"/templates/* "${TMP_DIR}"/.
+  cp -f -a "${WD}"/templates/* "${TMP_DIR}"/.
 }
 
-resetCluster() {
+function reset_cluster() {
   echo "Removing Istio CRDs"
   
   # Ideally we should use `istioctl x uninstall --purge -y`
   # But istioctl < 1.7 does not seem to support it. In order
   # not to make things complicated, I'm removing CRDs
   local istioctl=${1}
   "${istioctl}" x uninstall --purge -y
```

Comment:
to cover all options i suggest also adding
```
istioctl manifest generate | kubectl delete -f -
```
also, i suggest adding || true to the uninstall command above because it will fail in earlier versions

Reply:
Fixed it.